### PR TITLE
Drupal: New task table with navigation bar and application filter.

### PR DESCRIPTION
**Pull request by drshawnkwang**
New task table built without Views. Instead custom SQL queries are used to build a table using drupal's theme_table() function. The navigation bar is build with drupal's theme_item_list() function, with the class being 'tabs'.

For details see:
https://dev.gridrepublic.org/browse/DBOINCP-75
https://dev.gridrepublic.org/browse/DBOINCP-30

### CODE REVIEW

File: drupal/sites/default/boinc/modules/boincwork/boincwork.module [hunk 1]
```diff
@@ -162,6 +162,32 @@ function boincwork_menu() {
     'access arguments' => array('access content'),
     'type' => MENU_CALLBACK
   );
+  // @todo - change URL of routes
```
Comment (ChristianBeer) [inline drupal/sites/default/boinc/modules/boincwork/boincwork.module]: Is this still a todo? If yes, can you elaborate why and how the URLs should be changed?

File: drupal/sites/default/boinc/modules/boincwork/boincwork.module [hunk 2]
```diff
@@ -162,6 +162,32 @@ function boincwork_menu() {
...
+  $items['account/tasks/%/%'] = array(
+    'title' => 'Account Tasks Table',
```
Comment (ChristianBeer) [inline drupal/sites/default/boinc/modules/boincwork/boincwork.module]: Do those titles get translated somehow?

File: drupal/sites/default/boinc/modules/boincwork/boincwork.module [hunk 3]
```diff
@@ -973,6 +999,50 @@ function boincwork_user_wap() {
...
+  }
+
+  return boincwork_tasktable(2, $host_id, $tselect, $app_id);
+}
+
+/**
+ * Page callback for workunit task table
+ */
+function boincwork_workunit_task_table($workunit_id = NULL, $tselect = NULL, $app_id = NULL) {
+  $title = bts('Tasks for computer @workunit_id', array('@workunit_id' => $workunit_id), NULL, 'boinc:workunit-task-table');
```
Comment (ChristianBeer) [inline drupal/sites/default/boinc/modules/boincwork/boincwork.module]: Should read `Tasks for workunit @workunit_id`.

File: drupal/sites/default/boinc/modules/boincwork/includes/boincwork.helpers.inc [hunk 4]
```diff
@@ -1470,3 +1470,357 @@ function boincwork_task_time_reported($received_time = NULL, $deadline = NULL, $
...
+    array(
+      'data' => bts('Run time', array(), NULL, 'boinc:task-table'),
+      'field'=>'elapsed_time',
+    ),
+    array(
+      'data' => bts('CPU time', array(), NULL, 'boinc:task-table'),
+      'field'=>'cpu_time',
+    ),
+    array(
+      'data' => bts('Claimed Credit', array(), NULL, 'boinc:task-table'),
```
Comment (ChristianBeer) [inline drupal/sites/default/boinc/modules/boincwork/includes/boincwork.helpers.inc]: There is no notion of claimed credit in BOINC anymore. We should get rid of this and don't show it to the user.

File: drupal/sites/default/boinc/modules/boincwork/boincwork.module (hunk 1 above)
Comment (drshawnkwang) [inline drupal/sites/default/boinc/modules/boincwork/boincwork.module]: No. I changed the URL paths but did not remove the comment. I have now removed the comment.

File: drupal/sites/default/boinc/modules/boincwork/boincwork.module (hunk 2 above)
Comment (drshawnkwang) [inline drupal/sites/default/boinc/modules/boincwork/boincwork.module]: The Drupal API says they are to be untranslated.

File: drupal/sites/default/boinc/modules/boincwork/boincwork.module (hunk 3 above)
Comment (drshawnkwang) [inline drupal/sites/default/boinc/modules/boincwork/boincwork.module]: Thanks for catching this.

File: drupal/sites/default/boinc/modules/boincwork/includes/boincwork.helpers.inc (hunk 4 above)
Comment (brevilo) [inline drupal/sites/default/boinc/modules/boincwork/includes/boincwork.helpers.inc]: Did this get removed upstream? When and why? If so, then yes, let's sync up with BOINC again.

Comment (drshawnkwang) [inline drupal/sites/default/boinc/modules/boincwork/includes/boincwork.helpers.inc]: I think what we'll do is 1) merge this code with the claimed credit column. And 2) open a new PR that removes claimed credit across the site- it's in at least one other View, with a new JIRA issue ticket explaining what we're doing.

Comment (ChristianBeer) [inline drupal/sites/default/boinc/modules/boincwork/includes/boincwork.helpers.inc]: `claimed credit` is a remnant of the credit system before CreditNew. I concur with Shawns proposal, this way we can make sure we get every instance of this in the Drupal code.

File: drupal/sites/default/boinc/themes/boinc/templates/views-view-table--boinc-account-computers--panel-pane-1.tpl.php [hunk 5]
```diff
@@ -35,6 +35,6 @@
     <?php print l(bts('More', array(), NULL, 'boinc:account-dashboard'), 'account/computers'); ?>
   </li>
   <li class="first alt tab">
-    <?php print l(bts('Tasks', array(), NULL, 'boinc:host-tasks-link'), 'account/tasks/active'); ?>
+    <?php print l(bts('Tasks', array(), NULL, 'boinc:host-tasks-link'), 'account/tasks/0/1'); ?>
```
Comment (brevilo) [inline drupal/sites/default/boinc/themes/boinc/templates/views-view-table--boinc-account-computers--panel-pane-1.tpl.php]: The target `account/tasks/0/1` seems wrong. Shouldn't it be `account/tasks/0/0` instead?

Comment (drshawnkwang) [inline drupal/sites/default/boinc/themes/boinc/templates/views-view-table--boinc-account-computers--panel-pane-1.tpl.php]: Yes. In fact changing that link is now part of DBOINCP-71. I will generate a new PR today.

**Pull request merged**